nano-sync/src/primitives/pedersen.rs

```diff
@@ -0,0 +1,185 @@
+use algebra::mnt6_753::{Fq, G1Affine, G1Projective};
+use algebra_core::{Group, One, PrimeField};
+use blake2_rfc::blake2s::Blake2s;
+use crypto_primitives::prf::Blake2sWithParameterBlock;
+
+use crate::rand_gen::generate_random_seed;
+use crate::utils::big_int_from_bytes_be;
+
+/// This is the function for creating generators in the G1 group for the MNT6-753 curve. These
+/// generators are meant to be use for the Pedersen hash and the Pedersen commitment functions.
+pub fn pedersen_generators(number: usize) -> Vec<G1Projective> {
+    // This gets a verifiably random seed. Whenever we use this seed we need to set the personalization
+    // field on the Blake2X to an unique value. See below.
+    let seed = generate_random_seed();
+
+    // This extends the seed using the Blake2X algorithm.
+    // See https://blake2.net/blake2x.pdf for more details.
+    // We need 96 bytes of output for each generator that we are going to create.
+    // The number of rounds is calculated so that we get 96 bytes per generator needed.
+    let mut bytes = vec![];
+
+    let number_rounds = number * 3;
+
+    for i in 0..number_rounds {
+        let blake2x = Blake2sWithParameterBlock {
+            digest_length: 32,
+            key_length: 0,
+            fan_out: 0,
+            depth: 0,
+            leaf_length: 32,
+            node_offset: i as u32,
+            xof_digest_length: 65535,
+            node_depth: 0,
+            inner_length: 32,
+            salt: [1; 8],
+            // This needs to be set to an unique value, since we want a different random stream for
+            // each generator series that we create.
+            personalization: [0; 8],
```

Comment:
A lot of zeros don't look much unique to me. :)

Reply:
Well, unique in the sense that we want it to be different from the other generators. They all use the same seed and the same Blake2X, so we need the personalization field to be different.
But you know what? I'll set it to five. XD